moveit_ros/moveit_servo/src/servo_node.cpp

```diff
@@ -290,10 +293,14 @@ void ServoNode::servoLoop()
 
     if (next_joint_state && (servo_->getStatus() != StatusCode::INVALID))
     {
-      trajectory_publisher_->publish(composeTrajectoryMessage(servo_->getParams(), next_joint_state.value()));
+      if (servo_params_.command_out_type == "trajectory_msgs/JointTrajectory")
+        trajectory_publisher_->publish(composeTrajectoryMessage(servo_->getParams(), next_joint_state.value()));
+      else if (servo_params_.command_out_type == "std_msgs/Float64MultiArray")
+        multi_array_publisher_->publish(composeMultiArrayMessage(servo_->getParams(), next_joint_state.value()));
```

Comment:
This code addition is duplicated from above -- worth making it into a function?

Reply:
The one above is for publisher creation during `ServoNode` initialisation.
This is for publishing.

Reply:
wow, my bad!

I was confused because of the if/else. I think the publishers should both be created no matter what -- it doesn't hurt to have that extra publisher sitting around doing nothing, and would allow switching the command type at runtime (though not sure why someone would do that lol)

Reply:
In that case, we would have to add another `command_out_topic` parameter for the `Float64MultiArray`.
Otherwise two different type of publishers point to the same topic.
The old `servo_calcs` also use the [same](https://github.com/ros-planning/moveit2/blob/830ceda79b8f67c2224eb9ea8eb37f4aa0fe2596/moveit_ros/moveit_servo/src/servo_calcs.cpp#L109) condition based publisher creation

Reply:
Ah, I see, it's only one topic. Ignore me then, sorry!